app/src/main/java/com/google/samples/apps/sunflower/adapters/PlantAdapter.kt

```diff
@@ -23,27 +23,64 @@ import androidx.navigation.findNavController
 import androidx.recyclerview.widget.DiffUtil
 import androidx.recyclerview.widget.ListAdapter
 import androidx.recyclerview.widget.RecyclerView
+import androidx.recyclerview.widget.StaggeredGridLayoutManager
 import com.google.samples.apps.sunflower.PlantListFragment
+import com.google.samples.apps.sunflower.R
 import com.google.samples.apps.sunflower.ViewPagerFragmentDirections
 import com.google.samples.apps.sunflower.data.Plant
 import com.google.samples.apps.sunflower.databinding.ListItemPlantBinding
 
 /**
  * Adapter for the [RecyclerView] in [PlantListFragment].
  */
-class PlantAdapter : ListAdapter<Plant, PlantAdapter.ViewHolder>(PlantDiffCallback()) {
+class PlantAdapter : ListAdapter<Plant, RecyclerView.ViewHolder>(PlantDiffCallback()) {
 
-    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
-        val plant = getItem(position)
-        holder.apply {
-            bind(createOnClickListener(plant.plantId), plant)
-            itemView.tag = plant
+    private val headerPositions: IntArray = intArrayOf(0)
```

Comment:
Can remove type declaration.  Also add comments throughout the class describing how the header position is set up